js/script.js

```diff
@@ -0,0 +1,137 @@
+
+var gameState = 'notStarted';  //started // ended
+var player = {
+    name: '',
+    score: 0
+};
+var computer = {
+    score: 0
+};
+
+
+//Event listener
+newGameBtn.addEventListener('click', newGame); 
+pickRock.addEventListener('click', function() { playerPick('rock') }); 
+pickPaper.addEventListener('click', function() { playerPick('paper') }); 
+pickScissors.addEventListener('click', function() { playerPick('scissors') });
+
+
+//Wyświetlanie głównych elemntów strony: przycisku, tablicy wyboru i tablicy wyników
+function setGameElements() {
+    switch(gameState) {
+      case 'started':
+          newGameElem.style.display = 'none';
+          pickElem.style.display = 'block';
+          resultsElem.style.display = 'block';
+        break;
+      case 'ended':
+          newGameElem.style.display = 'block';
+          newGameBtn.innerText = 'Jeszcze raz';
+          resultsElem.style.display = 'block';
+          pickElem.style.display = 'none';
+        break;
+      case 'notStarted':
+      default:
+          newGameElem.style.display = 'block';
+          pickElem.style.display = 'none';
+          resultsElem.style.display = 'none';
+    }
+}
+
+setGameElements();
+
+//Liczenie punktów
+function setGamePoints() {
+    playerPointsElem.innerHTML = player.score;
+    computerPointsElem.innerHTML = computer.score;
+}
+
+//Rozpoczęcie gry
+function newGame() {
+    player.name = prompt('Please enter your name', 'imię gracza');
+    if (player.name) {
+      player.score = computer.score = 0;
+
+      playerPickElem.innerText = "Player selection";
+      computerPickElem.innerText = "Computer selection";
+      playerResultElem.innerText = "Player Score";
+      computerResultElem.innerText = "Computer Score";
+
+      pickElem.style.display = 'block';
+      winnerAlertElem.innerText = "";
+
+      gameState = 'started';
+      setGameElements();
+  
+      playerNameElem.innerHTML = player.name;
+      setGamePoints();
+    }
+}
+
+//Wybór gracza
+function playerPick(playerPick) {
+    var computerPick = getComputerPick();
+    
+    playerPickElem.innerHTML = playerPick;
+    computerPickElem.innerHTML = computerPick;
+
+    checkRoundWinner(playerPick, computerPick);
+}
+
+//Wybór komputera
+function getComputerPick() {
+    var possiblePicks = ['rock', 'paper', 'scissors'];
+    return possiblePicks[Math.floor(Math.random()*3)];
+}
+
+//Sprawdzenie kto wygrał rundę
+function checkRoundWinner(playerPick, computerPick) {
+    playerResultElem.innerHTML = computerResultElem.innerHTML = '';
+  
+    if (playerPick == computerPick) {
+        // remis
+    } else if (
+        (computerPick == 'rock' &&  playerPick == 'scissors') ||
+        (computerPick == 'scissors' &&  playerPick == 'paper') ||
+        (computerPick == 'paper' &&  playerPick == 'rock')) {
+  
+        computerResultElem.innerHTML = "Win!";
+        computer.score++;
+    } else {
+        playerResultElem.innerHTML = "Win!";
+        player.score++;
```

Comment:
Jest prawie ok, jeszcze wyrzuciłbym ogłaszanie wyniku do innej funkcji, przy okazji pozbywając się DRY z 119-123 linijki (bo sama przyznasz, że są mega podobne)

Stwórz funkcję `announceRoundWinner()` którą w tych dwóch miejscach odpalisz, tylko przekazując różne argumenty. Raz przekażesz result elem komputera i jego obiekt ze scorem, a raz result elem gracza i obiekt z jego scorem. Sama funkcja `announceRoundWinner` ma tylko dostać odpowiednie obiekty i je zaktualizować, niezależnie czy to będzie obiekt gracza czy komputera.

Odseparujemy wtedy logikę sprawdzania kto wygrał rundę od logiki wyświetlania informacji o zwycięzcy rundy, a kod nie będzie się powtarzał.